instrumentation/message-handler/aws/src/main/java/io/opentelemetry/instrumentation/messagehandler/SqsBatchMessageHandler.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.messagehandler;
+
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.common.AttributesBuilder;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import io.opentelemetry.instrumentation.api.instrumenter.AttributesExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.Instrumenter;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanLinksExtractor;
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+import io.opentelemetry.instrumentation.awssdk.v2_2.SqsMessageAccess;
+import io.opentelemetry.instrumentation.awssdk.v2_2.SqsParentContext;
+import io.opentelemetry.semconv.trace.attributes.SemanticAttributes;
+import java.util.Collection;
+import java.util.Map;
+import javax.annotation.Nullable;
+import software.amazon.awssdk.core.SdkPojo;
+import software.amazon.awssdk.services.sqs.model.Message;
+
+public abstract class SqsBatchMessageHandler extends BatchMessageHandler<Message> {
+  public SqsBatchMessageHandler(
+      OpenTelemetry openTelemetry, SpanNameExtractor<Collection<Message>> spanNameExtractor) {
+    super(openTelemetry, spanNameExtractor);
+  }
+
+  public SqsBatchMessageHandler(
+      OpenTelemetry openTelemetry,
+      SpanNameExtractor<Collection<Message>> spanNameExtractor,
+      String messageOperation) {
+    super(openTelemetry, spanNameExtractor, messageOperation);
+  }
+
+  @Override
+  protected void setup() {
+    messageInstrumenter =
+        Instrumenter.<Collection<Message>, Void>builder(
+                getOpenTelemetry(), "io.opentelemetry.message.handler", getSpanNameExtractor())
+            .setInstrumentationVersion("1.0")
```

Comment:
We never set the the instrumentation version manually but rely on it getting picked up automatically based on the instrumentation name. I guess you need to do this because your module name and the instrumentation name don't match? Perhaps you should make them match instead of hardcoding the version.

Reply:
Do I just need to change the name or do I need a class like the one below?

```
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.instrumentation.messagehandler;

import static java.util.Arrays.asList;

import com.google.auto.service.AutoService;
import io.opentelemetry.javaagent.extension.instrumentation.InstrumentationModule;
import io.opentelemetry.javaagent.extension.instrumentation.TypeInstrumentation;
import java.util.List;

@AutoService(InstrumentationModule.class)
public class SqsMessageHandlerInstrumentationModule extends InstrumentationModule {
  public SqsMessageHandlerInstrumentationModule() {
    super("message-handler", "message-handler-1.0");
  }

  @Override
  public List<TypeInstrumentation> typeInstrumentations() {
    return asList(
        {New instrumentation class here}
  }
}
```

Reply:
I cannot seem to get this to work without the version.

Reply:
You don't need an `InstrumentationModule`. When you don't specify a version it will be read from a file that is generated during the build based on the module name where your code is. The code that generates the file is in
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/5b03ae655fceba031e8ec147fb1438966e568837/conventions/src/main/kotlin/io.opentelemetry.instrumentation.base.gradle.kts#L140-L151 See what it generates in `build/generated/instrumentationVersion/META-INF/io/opentelemetry/instrumentation/` of your module. The file name there should give you the name you are looking for. My guess would be that it is `aws`. Our usual naming convention for your module would have been something like `message-handler-aws-2.2`